src/flag_gems/ops/isin.py

```diff
@@ -0,0 +1,259 @@
+import torch
+import triton
+import triton.language as tl
+
+from flag_gems.utils.libentry import libentry
+
+from .all import reduce_all
+from .any import reduce_any
+from .unique import _unique2
+
+
+def launch_arg(BLOCK_M, BLOCK_N, N, num_warps):
+    return BLOCK_M, min(BLOCK_N, triton.next_power_of_2(N)), num_warps
+
+
+@triton.jit
+def isin_by_comparation_impl(
+    global_pid,
+    in0_ravel_ptr: tl.tensor,
+    in1_ravel_ptr: tl.tensor,  # in
+    out_ptr: tl.tensor,  # out
+    M: int,  # num_tasks
+    N: int,  # num_tasks_1
+    BLOCK_M: tl.constexpr,  # tile_size
+    BLOCK_N: tl.constexpr,  # tile_size_1
+    invert: tl.constexpr,
+):
+    row_off = global_pid * BLOCK_M
+    rows = row_off + tl.arange(0, BLOCK_M)[:, None]
+    row_mask = rows < M
+    out_ptr += rows
+    in0_ravel_ptr += rows + tl.zeros([BLOCK_N], dtype=tl.int32)
+    in1_ravel_ptr += tl.zeros([BLOCK_M], dtype=tl.int32)[:, None]
+
+    block = tl.full([BLOCK_M, BLOCK_N], value=(1 if invert else 0), dtype=tl.int1)
+    in0 = tl.load(in0_ravel_ptr, row_mask, other=0)
+    for col_off in range(0, N, BLOCK_N):
+        cols = col_off + tl.arange(0, BLOCK_N)[None, :]
+        col_mask = cols < N
+        mask = row_mask and col_mask
+        in1 = tl.load(in1_ravel_ptr + cols, mask, other=0)
+        block = tl.where(
+            mask,
+            tl.where(invert, block and (in0 != in1), block or (in0 == in1)),
+            invert,
+        )
+    out = tl.reduce(block, axis=1, combine_fn=(reduce_all if invert else reduce_any))
+    tl.store(out_ptr, out[:, None], row_mask)
+
+
+@libentry()
+@triton.jit
+def isin_by_comparation_kernel(
+    in0_ravel_ptr: tl.tensor,
+    in1_ravel_ptr: tl.tensor,  # in
+    out_ptr: tl.tensor,  # out
+    M: int,  # num_tasks
+    N: int,  # num_tasks_1
+    BLOCK_M: tl.constexpr,  # tile_size
+    BLOCK_N: tl.constexpr,  # tile_size_1
+    tiles_per_cta: int,
+    invert: tl.constexpr,
+):
+    pid = tl.program_id(0)
+    num_ctas = tl.num_programs(0)
```

Comment:
num_ctas is a Triton reserved key word. We'd better rename it.

Reply:
> num_ctas is a Triton reserved key word. We'd better rename it.

done